### PR TITLE
Add metrics for current cache/object readers

### PR DESCRIPTION
**Pull request by polyrabbit**
Align with `object_request_uploading`, so we know the pressure we are exposing to the backend storage.

### CODE REVIEW

File: pkg/chunk/cached_store.go [hunk 1]
```diff
@@ -921,6 +926,20 @@ func (store *cachedStore) regMetrics(reg prometheus.Registerer) {
...
+		ConstLabels: prometheus.Labels{"storage": "cache"},
+	}, func() float64 {
+		return float64(store.cacheReaders.Load())
+	}))
+	reg.MustRegister(prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Name:        "store_readers",
+		Help:        "number of storage readers",
+		ConstLabels: prometheus.Labels{"storage": "object"},
+	}, func() float64 {
+		return float64(store.objectReaders.Load())
```
Comment (davies) [inline pkg/chunk/cached_store.go:941]: We can add a limit for concurrent GET, and know the actual concurrency using that.

File: pkg/chunk/cached_store.go [hunk 2]
```diff
@@ -921,6 +926,20 @@ func (store *cachedStore) regMetrics(reg prometheus.Registerer) {
 		func() float64 {
 			return float64(len(store.currentUpload))
 		}))
+	reg.MustRegister(prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Name:        "store_readers",
+		Help:        "number of storage readers",
+		ConstLabels: prometheus.Labels{"storage": "cache"},
+	}, func() float64 {
+		return float64(store.cacheReaders.Load())
```
Comment (davies) [inline pkg/chunk/cached_store.go:934]: We can know the concurrent IO from `dcState`

Comment (polyrabbit) [reply pkg/chunk/cached_store.go:934]: `concurrency` counter is only updated in `unstableDC` implementation. To track concurrency in other states, I have to move it to `baseDC`?

File: pkg/chunk/cached_store.go (hunk 1 above)
Comment (polyrabbit) [reply pkg/chunk/cached_store.go:941]: Currently we don't have that limitation right?

**Comment by davies** [comment]
We usually use the rate of object_request_durations_histogram_seconds as the metric of concurrency, no new metric needed.

Same to other operations or requests.

**Comment by polyrabbit** [comment]
OK, closing

**Pull request closed without merging**